File: 30daysOFcode_Python/day 06.py
'''
# Enter your code here. Read input from STDIN. Print output to STDOUT
t = int(input().strip())
for _ in range(t):
    s = input().strip()
    n = len(s)

    even = [s[i] for i in range(0, n, 2)] # list of even chars
    odd = [s[i] for i in range(1, n, 2)] # list of odd chars

    print(''.join(even), ''.join(odd))
'''

(lambda n, apply: [print(''.join(apply(input()))) for _ in range(n)    ]   )(   int(input()), lambda x : [x[i] for i in range(0, len(x), 2)] + [' '] + [x[i] for i in range(1, len(x), 2)]   )

File: 30daysOFcode_Go/solution_set2/day 06.go
package main

import "fmt"

func main() {
	//Enter your code here. Read input from STDIN. Print output to STDOUT
	// reading the number of string to be given
	var numberOfString int
	fmt.Scanln(&numberOfString)

	for i := 0; i < numberOfString; i++ {
		// reading the string in same variable again and again
		var inputString string
		fmt.Scanln(&inputString)

		for x := 0; x < len(inputString); x++ {
			//printing even position charaters
			if x%2 == 0 {
				// remember to convert to string because inputString[x] returns thier ASCII code/number
				fmt.Print(string(inputString[x]))
			}
		}
		fmt.Print(" ")
		for x := 0; x < len(inputString); x++ {
			//printing odd position charaters
			if x%2 != 0 {
				fmt.Print(string(inputString[x]))
			}
		}
		fmt.Print("\n")
	}
}
